vmm/sandbox/src/stratovirt/mod.rs

```diff
@@ -505,3 +499,18 @@ impl StratoVirtVM {
             .map_err(|e| Error::Other(anyhow!("start virtiofs daemon process failed: {}", e)))
     }
 }
+
+#[async_trait::async_trait]
+impl Recoverable for StratoVirtVM {
+    async fn recover(&mut self) -> Result<()> {
```

Comment:
There are repeated code in the `recover` function with different hypervisor type, maybe we can extract the same code block into trait function implementation.

Reply:
fixed